app/models/custom_field.rb

```diff
@@ -281,6 +285,14 @@ def multi_value_possible?
       [ProjectCustomField, WorkPackageCustomField].include?(self.class)
   end
 
+  def allow_non_open_versions?
+    allow_non_open_versions
+  end
+
+  def allow_non_open_versions_possible?
+    version? && [ProjectCustomField, WorkPackageCustomField].include?(self.class)
```

Comment:
There was no reason not to support it on the other types, but I missed the `TimeEntryCustomField` because I realized it supports version CF after this change. I'd add that. The support for `all_version` is actually there, at least in `app/contracts/time_entries/base_contract.rb` because I encountered the problem there while developing it. At this moment it is likely dormant, but I'd allow it there as well. There is on reason not to, if anyone finds a usage for it.
So I wanted to have it whenever Version CF can be - for the parity's sake - I just didn't know about time entries using it when I added it into that array (one of the first changes). I'll add it, if you don't have objection to it.

Reply:
One more note - the same problem is related to the code above for multi-value CFs (which is an omission from my previous PR #12948). I'll fix it both.